python/ray/train/gbdt_trainer.py

```diff
@@ -24,18 +24,37 @@ def _convert_scaling_config_to_ray_params(
     ray_params_cls: Type["xgboost_ray.RayParams"],
     default_ray_params: Optional[Dict[str, Any]] = None,
 ) -> "xgboost_ray.RayParams":
-    default_ray_params = default_ray_params or {}
-    resources_per_worker = scaling_config.additional_resources_per_worker
-    num_workers = scaling_config.num_workers
-    cpus_per_worker = scaling_config.num_cpus_per_worker
-    gpus_per_worker = scaling_config.num_gpus_per_worker
+    ray_params_kwargs = default_ray_params.copy() or {}
+
+    resources = scaling_config.resources_per_worker or {}
+
+    cpus_per_actor = resources.pop("CPU", 0)
+    if not cpus_per_actor:
+        cpus_per_actor = ray_params_kwargs.get("cpus_per_actor", 0)
+
+    gpus_per_actor = resources.pop("GPU", 0)
+    if not gpus_per_actor:
+        gpus_per_actor = default_ray_params.get("gpus_per_actor", 0)
+
+    resources_per_actor = resources
+    if not resources_per_actor:
+        resources_per_actor = default_ray_params.get("resources_per_actor", None)
+
+    num_actors = scaling_config.num_workers
+    if not num_actors:
+        num_actors = default_ray_params.get("num_actors", 0)
```

Comment:
Do we need to be using scaling config for these?

Reply:
I've added a docstring here - basically the default ray params are what we pass to the framework per default if nothing else is set (e.g. cpu per actor = 2 for lightgbm). But if the user specifies a scaling config where the CPUs per actor are defined, that should take precedence